### PR TITLE
refactor: convert channel_moderation files to ts

### PR DESCRIPTION
**Pull request by hereje**
#### Summary
refactor: convert channel_moderation files to ts

- convert js files to ts
- update data types accordingly
- fix lint issues

#### Ticket Link

Fixes: #21332

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: e2e-tests/cypress/tests/integration/channels/enterprise/system_console/channel_moderation/helpers.ts [hunk 1]
```diff
@@ -31,7 +34,7 @@ export const disablePermission = (permission) => {
 };
 
 // # Saves channel config and navigates back to the channel config page if specified
-export const saveConfigForChannel = (channelName = false, clickConfirmationButton = false) => {
+export const saveConfigForChannel = (channelName: string = null, clickConfirmationButton = false) => {
```
Comment (devinbinnie) [inline e2e-tests/cypress/tests/integration/channels/enterprise/system_console/channel_moderation/helpers.ts:37]: Was this always a `string`? Looks like it used to be a `boolean`, but we might have just been checking for a falsy case erroneously.

File: e2e-tests/cypress/tests/integration/channels/enterprise/system_console/channel_moderation/helpers.ts [hunk 2]
```diff
@@ -62,14 +65,14 @@ export const saveConfigForChannel = (channelName = false, clickConfirmationButto
 };
 
 // # Visits a channel as the member specified
-export const visitChannel = (user, channel, team) => {
-    cy.apiLogin(user);
+export const visitChannel = (user: User, channel: Channel, team: Team) => {
+    cy.apiLogin(user as UserProfile);
```
Comment (devinbinnie) [inline e2e-tests/cypress/tests/integration/channels/enterprise/system_console/channel_moderation/helpers.ts:69]: Can we just have the function take in a `UserProfile` if we're going to be immediately casting it?

**Review by devinbinnie — COMMENTED** [review]
Thanks for the PR @hereje! Just had a couple comments

File: e2e-tests/cypress/tests/integration/channels/enterprise/system_console/channel_moderation/helpers.ts (hunk 1 above)
Comment (hereje) [reply e2e-tests/cypress/tests/integration/channels/enterprise/system_console/channel_moderation/helpers.ts:37]: The variable name suggests a string; additionally that variable is used as input for the type function; see line 59

```ts
if (channelName) {
    // # Search for the channel.
    cy.get('.DataGrid_searchBar').within(() => {
        cy.findByPlaceholderText('Search').type(`${channelName}{enter}`); // here
    });
    cy.findByText('Edit').click();
}
```

File: e2e-tests/cypress/tests/integration/channels/enterprise/system_console/channel_moderation/helpers.ts (hunk 2 above)
Comment (hereje) [reply e2e-tests/cypress/tests/integration/channels/enterprise/system_console/channel_moderation/helpers.ts:69]: I'll change it to UserProfile

File: e2e-tests/cypress/tests/integration/channels/enterprise/system_console/channel_moderation/post_reactions_spec.ts [hunk 3]
```diff
@@ -43,15 +46,15 @@ describe('MM-23102 - Channel Moderation - Post Reactions', () => {
...
+            cy.apiCreateGuestUser({}).then(({guest}) => {
                 guestUser = guest;
 
                 cy.apiAddUserToTeam(testTeam.id, guestUser.id).then(() => {
                     cy.apiAddUserToChannel(testChannel.id, guestUser.id);
                 });
 
                 // Post a few messages in the channel
-                visitChannel(admin, testChannel, testTeam);
+                visitChannel(admin as UserProfile, testChannel, testTeam);
```
Comment (devinbinnie) [inline e2e-tests/cypress/tests/integration/channels/enterprise/system_console/channel_moderation/post_reactions_spec.ts:57]: So now the casting is moved out a bit, can we just have `getAdminAccount` return a `UserProfile`?

Comment (hereje) [reply e2e-tests/cypress/tests/integration/channels/enterprise/system_console/channel_moderation/post_reactions_spec.ts:57]: Yeah, sure.

File: e2e-tests/cypress/tests/integration/channels/enterprise/system_console/channel_moderation/helpers.ts (hunk 1 above)
Comment (pvev) [inline e2e-tests/cypress/tests/integration/channels/enterprise/system_console/channel_moderation/helpers.ts:37]: should this type be `string | null `

File: e2e-tests/cypress/tests/support/external_commands.ts [hunk 4]
```diff
@@ -13,7 +13,7 @@ import {getRandomId} from '../utils';
 function externalActivateUser(userId: string, active = true) {
     const admin = getAdminAccount();
 
-    cy.externalRequest({user: admin, method: 'PUT', path: `users/${userId}/active`, data: {active}});
+    return cy.externalRequest({user: admin, method: 'PUT', path: `users/${userId}/active`, data: {active}});
```
Comment (pvev) [inline e2e-tests/cypress/tests/support/external_commands.ts:16]: https://github.com/mattermost/mattermost/pull/28483#discussion_r1790001474

**Review by pvev — APPROVED** [review]
LGTM! just some minor questions. Thanks @hereje 🎉

**Comment by devinbinnie** [comment]
@saturninoabril Pinging cause it's E2E related, if we're good with 2 reviewers then feel free to just merge :)

**Review by saturninoabril — APPROVED** [review]
Thanks @hereje, LGTM!

**Pull request merged**